checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,363 @@
+          }
+        }
+      }
+    }
+  }
+
+  @Override
+  public AnnotatedTypeMirror fromElement(Element elt) {
+    AnnotatedTypeMirror type = super.fromElement(elt);
+    // Support @InheritableMustCall meaning @MustCall on all class declaration elements.
+    if (ElementUtils.isTypeElement(elt)) {
+      AnnotationMirror inheritableMustCall = getDeclAnnotation(elt, InheritableMustCall.class);
+      if (inheritableMustCall != null) {
+        List<String> mustCallVal =
+            AnnotationUtils.getElementValueArray(
+                inheritableMustCall, inheritableMustCallValueElement, String.class);
+        AnnotationMirror inheritedMCAnno = createMustCall(mustCallVal.toArray(new String[0]));
+        // Ensure that there isn't an inconsistent, user-written @MustCall annotation and
+        // issue an error if there is. Otherwise, replace the implicit @MustCall({}) with
+        // the inherited must-call annotation.
+        AnnotationMirror writtenMCAnno = type.getAnnotationInHierarchy(this.TOP);
+        if (writtenMCAnno != null
+            && !this.getQualifierHierarchy().isSubtype(inheritedMCAnno, writtenMCAnno)) {
+          if (!elementsIssuedInconsistentMustCallSubtypeErrors.contains(elt)
+              && !this.checker.shouldSkipUses(elt)) {
+            checker.reportError(
+                elt,
+                "inconsistent.mustcall.subtype",
+                elt.getSimpleName(),
+                writtenMCAnno,
+                inheritableMustCall);
+            elementsIssuedInconsistentMustCallSubtypeErrors.add(elt);
+          }
+        } else {
+          type.replaceAnnotation(inheritedMCAnno);
+        }
+      }
+    }
+    return type;
+  }
+
+  /**
+   * Creates a {@link MustCall} annotation whose values are the given strings.
+   *
+   * @param val the methods that should be called
+   * @return an annotation indicating that the given methods should be called
+   */
+  public AnnotationMirror createMustCall(final String... val) {
+    AnnotationBuilder builder = new AnnotationBuilder(processingEnv, MustCall.class);
+    Arrays.sort(val);
+    builder.setValue("value", val);
+    return builder.build();
+  }
+
+  @Override
+  public QualifierHierarchy createQualifierHierarchy() {
+    return new SubtypeIsSubsetQualifierHierarchy(
+        this.getSupportedTypeQualifiers(), this.getProcessingEnv());
+  }
+
+  /**
+   * Fetches the store from the results of dataflow for block. If useBlock is true, then the store
+   * after block is returned; if useBlock is false, the store before succ is returned.
+   *
+   * @param useBlock whether to use the store after the block itself or the store before its
+   *     successor, succ
+   * @param block a block
+   * @param succ block's successor
+   * @return the appropriate CFStore, populated with MustCall annotations, from the results of
+   *     running dataflow
+   */
+  public CFStore getStoreForBlock(boolean useBlock, Block block, Block succ) {
```

Comment:
This method is not used.  Is it needed?  Even if needed, can it be inlined at its future call sites?

If you retain it, I would rename `useBlock` to `afterFirstStore` or the like, to better communicate what it means.

Reply:
> can it be inlined at its future call sites

No, because `flowResult` is `protected`.

I changed the name as you suggested.